src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -40355,6 +42000,27 @@ static void F(S x, S? y)
             comp.VerifyDiagnostics();
         }
 
+        [Fact]
+        public void BinaryOperator_15_WithDisallowNull()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+struct S
+{
+    public static S operator+(S a, S b) => a;
+    static void F(S x, [DisallowNull] S? y)
+    {
+        S? s;
+        s = x + x;
+        s = x + y;
+        s = y + x;
+        s = y + y;
+    }
+}";
```

Comment:
What are we testing in this case? Aren't the `+` operations involving `y` operations on `S?` rather than `S`?

Reply:
My bad. Meant to test [DisallowNull] on the operator itself.

---
In reply to: [295080172](https://github.com/dotnet/roslyn/pull/36444#discussion_r295080172) [](ancestors = 295080172)